Objective-C/Internal/Replicator/CBLReachability.m

```diff
@@ -205,9 +205,11 @@ - (NSString*) description {
 
 - (BOOL) reachable {
     // We want 'reachable' flag to be on, but not if user intervention is required (like PPP login)
-    return _reachabilityKnown
-        &&  (_reachabilityFlags & kSCNetworkReachabilityFlagsReachable)
-        && !(_reachabilityFlags & kSCNetworkReachabilityFlagsInterventionRequired);
+    SCNetworkReachabilityFlags flag;
+    if (SCNetworkReachabilityGetFlags(_ref, &flag))
```

Comment:
I think if the _rechabilityKnow is true, we can just use _reachabilityFlags. You will need to change the implementation of the status method as well.

Reply:
BTW, The CBLReachability is not thread-safe by its implementation which is fine as long as the CBLReplicator doesn't call it from multiple threads (please double check as well).

Reply:
all the `_reachability` instances are called from lock.